src/index.js

```diff
@@ -1,9 +1,36 @@
-var core = module.exports = require('./core');
-
-// plugins:
-core.extras         = require('./extras');
-core.filters        = require('./filters');
-core.interaction    = require('./interaction');
-core.loaders        = require('./loaders');
-core.spine          = require('./spine');
-core.text           = require('./text');
+var core = require('./core'),
+	math = core.math,
+	// plugins:
+	extras         = require('./extras'),
+	filters        = require('./filters'),
+	interaction    = require('./interaction'),
+	loaders        = require('./loaders'),
+	spine          = require('./spine'),
+	text           = require('./text'),
+    deprecation    = require('./deprecation');
+
+var modules = [];
+
+modules.push(math);
+modules.push(extras);
+modules.push(filters);
+modules.push(interaction);
+modules.push(loaders);
+modules.push(spine);
+modules.push(text);
+
+for (var i = 0; i < modules.length; i++) {
+
+	// Adds the module to the core, so you can do
+	// PIXI.Text instead of PIXI.text.Text
+	bundleUp(core,modules[i]);
+};
+
+function bundleUp(object1,object2){
+
+    for(var keys in object2){
+        object1[keys] = object2[keys];
+    }
+}
+
+module.exports = core;
```

Comment:
I really liked the namespacing, when did we decide to do away with it? I made it easy to understand the plugin system and what things are optional and what are plugins.

Even if we were to do it this way, why build an array manually then loop through it? Why not just call `bundleUp` on the require directly?